src/lib/data/html/speeddial.html

```diff
@@ -903,19 +988,68 @@
 }
 </script>
 </head>
-
 <body>
 
 <noscript>%JAVASCRIPT-DISABLED%</noscript>
 
-<div id="quickdial">
+<div class="SearchBar" id="SearchBar">
+    <div class="SearchBar" id="SearchBarToggle" onClick="searchToggle()">
+    </div>
+        <div id="SearchForm">
+            <input id="SearchInput" name="" value="%SEARCH-HINT%" onfocus="searchFocus()" onfocusout="searchRelease()">
+            <input id="submit" type="button" value="" onClick="Search()">
+        </div>
+     <div style="width: 36px"></div>
 </div>
 
-<a onClick="configureSpeedDial()" title="%SETTINGS-TITLE%" class="sett"></a>
+<a onClick="configureSpeedDial()" title="%SETTINGS-TITLE%" class="sett" id="sett"></a>
 <a onClick="addSpeedDial()" title="%ADD-TITLE%" class="add"></a>
+<div id="quickdial">
+</div>
+
 
 <script type="text/javascript">
 
+if ("%IMG_BACKGROUND%" == '') {
+    document.getElementById("html").style.background = 'url("qrc:html/sd_bg.svg")';
+    document.getElementById("html").style.backgroundSize = 'cover';
+}
+
+if (!%SEARCH-ENABLED%) {
+    document.getElementById('quickdial').style.margin = '10px auto';
+    document.getElementById('SearchBar').style.width = '40px';
+    document.getElementById('SearchBar').style.background = 'rgba(0,0,0, 0.0)';
+    document.getElementById('sett').style.top = '4px';
+    document.getElementById('SearchBarToggle').style.background = 'url("qrc:html/search.svg") no-repeat center center, rgba(65,65,65, 0.0)';
+    document.getElementById('SearchForm').style.visibility = 'hidden';
+}
+
+function searchFocus() {
+    document.getElementById('SearchInput').style.color = 'rgba(236,236,236, 1.0)';
```

Comment:
This should be in stylesheet

Reply:
Yep, but second value need to be changed here, and i see no reason to split them to different code blocks.
There is no problem to move this to stylesheet, but why it needed?

Reply:
Because style should be separate from logic.
Also, as those functions will then be just one statement, you can delete them and move the code inline in html tag.